src/zeit/content/volume/browser/toc.py

```diff
@@ -0,0 +1,392 @@
+# -*- coding: utf-8 -*-
+import StringIO
+import csv
+import tinydav
+import re
+import sys
+import lxml.etree
+import posixpath
+from ordereddict import OrderedDict
+import zeit.cms.browser.view
+from tinydav.exception import HTTPUserError
+from zeit.cms.i18n import MessageFactory as _
+import zeit.cms.content.sources
+import toc_config
+
+# TODO Author/Title Teaser contains is ugly
+# TODO More Abstraction of Toc Entries, not sure if an ordered dict is a great idea
+
+class Toc(zeit.cms.browser.view.Base):
+    """
+    View for creating a Table of Content as a csv file.
+    """
+    # Not all directories in the archives are named the product ID.
+    PRODUCT_ID_DIR_NAME_EXCEPTIONS = {'CW': 'ZECW'}
+    CSV_DELIMITER = '\t'
+
+    def __init__(self, *args, **kwargs):
+        super(Toc, self).__init__(*args, **kwargs)
+        self.client = self._create_dav_client()
+        self.article_excluder = ArticleExcluder()
+
+    def __call__(self):
+        self.volume = self.context
+        self.product_id_mapping = self._create_product_id_full_name_mapping()
+        filename = self._generate_file_name()
+        self.request.response.setHeader('Content-Type', 'text/csv')
+        self.request.response.setHeader('Content-Disposition', 'attachment; filename="%s"' % filename)
```

Comment:
Wir versuchen, uns an PEP8 zu halten (dazu gehört z.B. die maximale Zeilenlänge). In Friedbert ist das sogar Teil vom Testrunner (in vivi müsste ich es mal einbauen), in jedem Fall empfiehlt es sich, das als Plugin in seinem Editor zu aktivieren.